quantity/tests/test_operations.py

```diff
@@ -0,0 +1,293 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Test that operations on Quantity properly propagate units.
+
+Note: tests classes are combined with setups for different array types
+at the very end.  Hence, they do not have the usual Test prefix.
+"""
+
+from __future__ import annotations
+
+import operator
+
+import astropy.units as u
+import numpy as np
+import pytest
+from numpy.testing import assert_array_almost_equal_nulp, assert_array_equal
+
+from quantity import Quantity
+
+from .conftest import ARRAY_NAMESPACES
+
+
+def assert_quantity_equal(q1, q2, nulp=0):
+    assert q1.unit == q2.unit
+    assert type(q1.value) is type(q2.value)
+    assert_array_almost_equal_nulp(q1.value, q2.value, nulp=nulp)
+
+
+class QuantitySetup:
+    @classmethod
+    def setup_class(cls):
+        super().setup_class()
+        cls.a1 = cls.xp.asarray(np.arange(1.0, 11.0).reshape(5, 2))
+        cls.a2 = cls.xp.asarray([8.0, 10.0])
+        cls.q1 = Quantity(cls.a1, u.meter)
+        cls.q2 = Quantity(cls.a2, u.centimeter)
+
+
+class QuantityOperationTests(QuantitySetup):
+    def test_addition(self):
+        # Take units from left object, q1
+        r = self.q1 + self.q2
```

Comment:
nitpick, but 
```suggestion
        got = self.q1 + self.q2
```
has the same length as `exp`. 🙃

Reply:
Done